vpr/src/route/route_path_manager.h

```diff
@@ -0,0 +1,97 @@
+#include "rr_graph_fwd.h"
+#include "vtr_assert.h"
+
+#include <set>
+#include <list>
+#include <vector>
+
+#ifndef _PATH_MANAGER_H
+#    define _PATH_MANAGER_H
+
+/* Extra path data needed by RCV, seperated from t_heap struct for performance reasons
+ * Can be accessed by a pointer, won't be initialized unless needed
+ *
+ * path_rr: The entire partial path up until the route tree
+ * 
+ * edge: A list of edges from each node in the partial path to reach the next node
+ * 
+ * backward_delay: The delay of the partial path plus the path from route tree to source
+ * 
+ * backward_cong: The congestion estimate of the partial path plus the path from route tree to source */
+struct t_heap_path {
+    std::vector<RRNodeId> path_rr;
+    std::vector<RREdgeId> edge;
+    float backward_delay = 0.;
+    float backward_cong = 0.;
+};
+
+// Forward declaration of RoutingContext needed for traceback insertion
+class RoutingContext;
+
+/* A class to manage the extra data required for RCV
+ * It manages a set containing all the nodes that currently exist in the route tree
+ * This class also manages the extra memory allocation required for the t_heap_path structure */
+class PathManager {
+  public:
+    PathManager();
+    ~PathManager();
+
+    // Checks if the target node exists in the route tree, or current backwards path variable
```

Comment:
Do you have a comment that explains how RCV can create loops, and that two tests are necessary to prevent loops (check the partial route created for other connections on the net (route_tree_set) and check the prior nodes on the path that is being expanded/explored for a given heap element (partial path) for the current connection. This latter part needs to be stored for every partial path (heap element). 

If you don't have such an overview comment, you should add it, probably near the top of route_path_manager.h.